packages/library-base/build.gradle.kts

```diff
@@ -135,6 +135,12 @@ kotlin {
 //                .all { onlyIf { findProperty("isMainHost") == "true" } }
 //        }
 //    }
+
+    // Require that all methods in the API have visibility modifiers and return types.
+    // Anything inside `io.realm.internal.*` is considered internal regardless of their
+    // visibility modifier and will be stripped from Dokka, but will unfortunately still
+    // leak into auto-complete in the IDE.
```

Comment:
That is weird because I specifically tested this and it wasn't the case. Maybe this has changed in the past year or maybe my experiments was contaminated by some other KMP/gradle issues (not that we haven't seen red lines, etc. over the past year). We can maybe `@Suppress("invisible_reference", "invisible_member")` our way through internalizing more symbols from `library-base` even though they are used in `library-sync`, but I think there are issues if the type is extended.

Reply:
Not sure there are problems with extensions as such, but interface delegation across module boundaries is definitely broken. I should probably try to create a minimal project we can throw at JetBrains.

Alternatively we should rethink our current module setup in order to work around these issues. I'll create an issue that capture this.